### PR TITLE
refactor: fix partner artist list

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [APPL-313] 

### Description

[slack thread](https://artsy.slack.com/archives/C9SATFLUU/p1677075697155079) for context

#### Demo

https://user-images.githubusercontent.com/21178754/220868133-25e114e4-16fa-4e05-9b84-97e65342c34b.mp4

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[APPL-313]: https://artsyproduct.atlassian.net/browse/APPL-313?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Partner/Components/PartnerOverview.tsx [hunk 1]
```diff
@@ -41,49 +27,26 @@ export const PartnerOverview: React.FC<{
...
-        <>
-          <Text>
-            <Text variant="sm-display">Artists ({artistsData.length})</Text>
-          </Text>
-          <Spacer y={2} />
-          {renderArtists()}
-          <Spacer y={4} />
-        </>
-      )}
+      {!!displayArtistsSection ? <PartnerArtistsList partner={partner} /> : null}
```
Comment (gkartalis) [inline src/app/Scenes/Partner/Components/PartnerOverview.tsx:35]: this is not enough to fix the issue, even when we skip rendering the component it crashes still. The reason is [this line](https://github.com/artsy/eigen/pull/8254/files#diff-3ddeebbd8bdf711464e54166fa91394d6dc90eaa2317292edd6d0fd8ae7be278R49). 

What we need to do in order to fix it is to **skip** with a way this fragment from being registered if the `displayArtistsSection = false`

File: src/app/Scenes/Partner/Partner.tsx [hunk 2]
```diff
@@ -81,12 +81,6 @@ export const PartnerContainer = createRefetchContainer(
         slug
         partnerType
         displayFullPartnerPage
-        profile {
```
Comment (gkartalis) [inline src/app/Scenes/Partner/Partner.tsx:84]: just removed some stuff that weren't used anywhere

File: src/app/Scenes/Partner/Components/PartnerOverview.tsx [hunk 3]
```diff
@@ -41,49 +27,26 @@ export const PartnerOverview: React.FC<{
...
-            id
-            ...ArtistListItem_artist
-            counts {
-              artworks
-            }
-          }
-        }
-      }
       ...PartnerLocationSection_partner
+      ...PartnerArtistsList_partner
```
Comment (gkartalis) [inline src/app/Scenes/Partner/Components/PartnerOverview.tsx:49]: if you comment this out, the app stops crashing. 

The error looks like this:
simulator_screenshot_7EAC073F-7A04-4D0C-B1F5-F4405A5EAC69

File: src/app/Scenes/Partner/Components/PartnerArtistsList.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  return (
+    <>
+      <Text variant="sm-display">Artists ({allArtistsConnection?.totalCount})</Text>
+      <Spacer y={2} />
+      <FlatList
+        data={artists}
+        keyExtractor={(item) => item.id}
+        renderItem={({ item }) => <ArtistListItem artist={item} />}
+        ItemSeparatorComponent={() => <Spacer y={2} />}
+        initialNumToRender={15}
```
Comment (gkartalis) [inline src/app/Scenes/Partner/Components/PartnerArtistsList.tsx:36]: not sure if this is actually doing something

File: src/app/Scenes/Partner/Partner.tsx [hunk 5]
```diff
@@ -108,20 +105,39 @@ export const PartnerQueryRenderer: React.FC<{
...
+          displayArtistsSection
+        }
+      }
+    `,
+    variables: { partnerID },
+  })
+
+  if (loading) {
+    return <HeaderTabsGridPlaceholder />
+  }
```
Comment (damassi) [inline src/app/Scenes/Partner/Partner.tsx:122]: Might want to play with this a bit but I couldn't figure out how to thread this into the query without two fetches.

File: src/app/utils/useClientQuery.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * A more functional version of `fetchQuery`.
+ *
+ * Note! The data returned from this hook can be garbage collected at any time
+ * so you might need to call `retain` on it.
+ *
+ * @see https://relay.dev/docs/api-reference/fetch-query/#behavior
+ */
+
+export const useClientQuery = <T extends OperationType>({
```
Comment (damassi) [inline src/app/utils/useClientQuery.ts:14]: Copied this handy hook over from force

**Review by lordkiz — APPROVED** [review]
Thanks

**Pull request merged**